### DISCUSSION TITLE
What's the best way to update the device tree?

### DISCUSSION THREAD

**Discussion by Achder** [OE4T/meta-tegra#934] [Q&A]
I'm currently debugging issues with my device tree.
With each change I adjust my linux-tegra patch, recompile the image and flash it to the device. 
Is there a better way to do this?

**Comment by ichergui** [comment]
Hey @Achder 

You could try using `$ ./doflash.sh -c "sign; write DTB signed/DTB-FILE-NAME.dtb.signed"`

**Reply by Achder** [reply]
Could you elaborate on the whole workflow?

**Reply by ichergui** [reply]
Yes, 
After finishing the build of your image, you can use `doflash.sh` script from `tegraflash.tar.gz` package to flash only the `dtb`

**Reply by Achder** [reply]
Is there a way to create the dtb without building the whole image? The image compression at the end takes ~20 mins

**Reply by ichergui** [reply]
You can build the Linux kernel `linux-tegra` separately and use `doflash.sh` script (from `tegraflash.tar.gz`) or `flash.sh` from Jetpack to flash your `DTB`
